model/host/hostutil.go

```diff
@@ -276,32 +287,106 @@ func (h *Host) writeJasperCredentialsFileCommand(config evergreen.HostJasperConf
 }
 
-// RunSSHJasperRequest runs the command to make a request to the host's
-// Jasper service over SSH. It invoking the subcommand subCmd with the given
-// request input.
-func (h *Host) RunSSHJasperRequest(ctx context.Context, env evergreen.Environment, subCmd string, input interface{}, sshOptions []string) (string, error) {
-	inputBytes, err := json.Marshal(input)
+// RunJasperProcess makes a request to the host's Jasper service to run the
+// given command, wait for its completion, and return the output from it.
+func (h *Host) RunJasperProcess(ctx context.Context, env evergreen.Environment, sshOpts []string, command string) (string, error) {
+	client, err := h.JasperClient(ctx, env, sshOpts)
 	if err != nil {
-		return "", err
+		return "", errors.Wrap(err, "could not get a Jasper client")
 	}
-
-	config := env.Settings().HostJasper
-	binaryPath := h.jasperBinaryFileName(config)
-
 	output := &util.CappedWriter{
 		Buffer:   &bytes.Buffer{},
-		MaxBytes: 1024 * 1024, // 1MB
+		MaxBytes: 1024 * 1024, // 1 MB
 	}
 
-	hostInfo, err := h.GetSSHInfo()
+	splitCmd, err := shlex.Split(command)
 	if err != nil {
-		return "", errors.WithStack(err)
+		return "", errors.Wrap(err, "problem splitting command")
+	}
+	opts := jasper.CreateOptions{Args: splitCmd}
+	proc, err := client.CreateProcess(ctx, &opts)
+	if err != nil {
+		return "", errors.Wrap(err, "problem creating process")
 	}
 
-	err = env.JasperManager().CreateCommand(ctx).Host(hostInfo.Hostname).User(hostInfo.User).
-		ExtendSSHArgs("-p", hostInfo.Port, "-t", "-t").ExtendSSHArgs(sshOptions...).
-		SetOutputWriter(output).RedirectErrorToOutput(true).
-		Append(fmt.Sprintf("%s jasper client %s --service=rpc --port=%d <<EOF\n%s\nEOF", binaryPath, subCmd, config.Port, inputBytes)).
-		Run(ctx)
-	return output.String(), errors.Wrap(err, "error making Jasper request over SSH")
+	exitCode, err := proc.Wait(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "problem waiting for process completion")
+	}
+	if exitCode != 0 {
+		return "", errors.Errorf("process returned exit code %d", exitCode)
+	}
+
+	return output.String(), nil
+}
+
+// StartJasperProcess makes a request to the host's Jasper service to start a
+// process with the given options without waiting for its completion.
+func (h *Host) StartJasperProcess(ctx context.Context, env evergreen.Environment, sshOpts []string, opts *jasper.CreateOptions) error {
+	client, err := h.JasperClient(ctx, env, sshOpts)
+	if err != nil {
+		return errors.Wrap(err, "could not get a Jasper client")
+	}
+
+	if _, err := client.CreateProcess(ctx, opts); err != nil {
+		return errors.Wrap(err, "problem creating Jasper process")
+	}
+
+	return nil
+}
+
+// JasperClient returns a remote client that communicates with this host's
+// Jasper service.
+func (h *Host) JasperClient(ctx context.Context, env evergreen.Environment, sshOpts []string) (jasper.Manager, error) {
```

Comment:
do we have to pass sshopts to this?

Reply:
I changed this so that host has a `GetSSHOptions()` method that copies the EC2 cloud manager implementation of this (which is the same for all providers currently). I also added a ticket to improve the handling of SSH keys given by `GetSSHOptions()`.